lib/twilio/device.ts

```diff
@@ -1223,6 +1223,16 @@ class Device extends EventEmitter {
         // Stop trying to register presence after token expires
         this._stopRegistrationTimer();
         twilioError = new AuthorizationErrors.AccessTokenExpired(originalError);
+      } else if (code === 31209 ) {
+        this._publisher.error('call-message', 'error', {
+          code,
+          message: customMessage,
+        }, this._activeCall);
+      } else if (code === 31100 ) {
+        this._publisher.error('call-message', 'error', {
+          code,
+          message: customMessage,
+        }, this._activeCall);
```

Comment:
If we only check for specific error codes, we will miss other future errors unless we update the code. Let's make this dynamic. I suggest use this [if block](https://github.com/twilio/twilio-voice.js/blob/feature/eventing_1.5/lib/twilio/call.ts#L1429). We already check in there if the error is related to call message. You can use the payload object to read the information you need to send to insights. Also, we need to send `voice_event_sid` as well. It's missing in your code above. See schema here https://code.hq.twilio.com/client/rtc-insights-spec/blob/master/schema/voice/event/call-message/error/index.json

Reply:
The schema has it as "not required". It looks like from the Kibana logs it is getting auto populated. Do you still want me to add it in?

Reply:
Are you talking about voice event sid or callsid?

Reply:
`voice_event_sid` it is not in the "required" section: https://code.hq.twilio.com/client/rtc-insights-spec/blob/master/schema/voice/event/call-message/error/index.json#L35